### ISSUE TITLE
"named-parameters-mapping" should not apply to nested mappings

### ISSUE THREAD

**Issue by PaulRBerg** [protofire/solhint#408]
I have recently [refactored](https://github.com/PaulRBerg/prb-proxy/blob/aaaf577d2452c870d88de8f79167f588e4be4ab8/src/PRBProxy.sol#L40-L41) my PRBProxy project to use named mapping parameters, and then I tried to enable the new Solhint rules `named-parameters-mapping`. But I got this warning:

> warning  Value parameter in mapping permissions is not named  named-parameters-mapping

This is in spite of the fact that all params are named. It's just that the nested mapping themselves are not named, but I don't want to do this, because this doesn't make sense:

```
mapping(address envoy => mapping(address target => mapping(bytes4 selector => bool permission))) internal permissions;
```

**Comment by dbale-altoros** [comment]
Ok, I will review this one I think the two nested mapping inside one is causing the false positive... 

But regarding the other comment...
What do you mean it doesn't make sense to name the nested ones ?
This example names the nested mapping and in my opinion it does make sense
`mapping(address owner => mapping(address token => uint256 balance)) public tokenBalances`
Maybe I misunderstood your point ?

**Comment by PaulRBerg** [comment]
Thanks, Diego!

> What do you mean it doesn't make sense to name the nested ones ?

Let's take a simpler example:

```solidity
mapping(uint foo => mapping(uint bar => uint256 baz));
```

The above is totally fine. What is not fine is to name the nested mapping itself, like this:

```solidity
mapping(uint foo => mapping(uint bar => uint256 baz) nestedMapping);
```

The code above compiles but it doesn't make sense to define the nested mapping.

**Comment by fvictorio** [comment]
Also: I think the second snippet that Paul shared shouldn't be enforced, but it shouldn't be forbidden either. I don't think anyone will add names there, but forbidding it doesn't seem necessary.

**Comment by PaulRBerg** [comment]
Yup, agree; not enforcing but not forbidding either sounds like the best approach to take here.

**Comment by dbale-altoros** [comment]
Really ? The point of this addition it is just giving information about the enter key to the mapping and what the mapping contains in each "cell" on top of the mapping name... 
This example that was mention (@PaulRBerg)
`mapping(uint foo => mapping(uint bar => uint256 baz));`
Is missing the mapping name, so what do you mean by 'it is totally fine' ?... 

So, as I understand, you agree that this information only makes sense in the MAIN mapping, but it is useless in the nested ones ?
`mapping(address owner => mapping(address token => uint256 balance)) public tokenBalances`
This one shows info about the main enter key, the second enter key and the content... Don't see anything that doesn't make sense here... At most it is a bit 'over explained' ?
But it's ok. You guys are here long before me and your approach is simpler in terms of coding... I can fix it that way...

So if the rule is ON:
Will check only the naming of the KEY of the MAIN mapping and give an error if it is not present
And left alone the other named parameters on the mapping (doesn't matter if they exist or not)... no report of them...
Is that OK ?
@fvictorio

**Comment by PaulRBerg** [comment]
I was mostly thinking about nested mappings; I think the naming should still be enforced for values because they may not be inferable from the name of the parent mapping.

**Comment by dbale-altoros** [comment]
ok , so to be clear:

For regular mapping, this is the only way to write them and both variable naming should be enforced:
` mapping(address token => uint256 balance) public tokenBalances;`

For nested mapping
- The MAIN KEY variable (the entrance of the main mapping) should be named and this is enforced... 
- Every other variable can be named or not... this is not enforced 

Agree ?

**Comment by PaulRBerg** [comment]
Yes.

**Comment by dbale-altoros** [comment]
fixed on https://github.com/protofire/solhint/pull/421
